server/internal/infrastructure/db/sqlite/round_repo.go

```diff
@@ -104,50 +122,25 @@ ON CONFLICT(payment_id, pubkey) DO UPDATE SET
-FROM round 
-LEFT OUTER JOIN payment ON round.txid=payment.txid 
-LEFT OUTER JOIN receiver ON payment.id=receiver.payment_id
-LEFT OUTER JOIN vtxo ON payment.id=vtxo.payment_id
-WHERE round.ended = false AND round.failed = false
-`
-
-	selectRoundWithId = `
-SELECT round.id, round.starting_timestamp, round.ending_timestamp, round.ended, round.failed, round.stage_code, round.txid, round.unsigned_tx, round.congestion_tree, round.forfeit_txs, round.connectors, round.connector_address, round.dust_amount, round.version, round.swept, payment.id, receiver.payment_id, receiver.pubkey, receiver.amount, receiver.onchain_address, vtxo.txid, vtxo.vout, vtxo.pubkey, vtxo.amount, vtxo.pool_tx, vtxo.spent_by, vtxo.spent, vtxo.redeemed, vtxo.swept, vtxo.expire_at, vtxo.payment_id
-FROM round 
-LEFT OUTER JOIN payment ON round.txid=payment.txid 
-LEFT OUTER JOIN receiver ON payment.id=receiver.payment_id
-LEFT OUTER JOIN vtxo ON payment.id=vtxo.payment_id
-WHERE round.id = ?;
-`
-
-	selectRoundWithTxId = `
-SELECT round.id, round.starting_timestamp, round.ending_timestamp, round.ended, round.failed, round.stage_code, round.txid, round.unsigned_tx, round.congestion_tree, round.forfeit_txs, round.connectors, round.connector_address, round.dust_amount, round.version, round.swept, payment.id, receiver.payment_id, receiver.pubkey, receiver.amount, receiver.onchain_address, vtxo.txid, vtxo.vout, vtxo.pubkey, vtxo.amount, vtxo.pool_tx, vtxo.spent_by, vtxo.spent, vtxo.redeemed, vtxo.swept, vtxo.expire_at, vtxo.payment_id
-FROM round 
-LEFT OUTER JOIN payment ON round.txid=payment.txid 
-LEFT OUTER JOIN receiver ON payment.id=receiver.payment_id
-LEFT OUTER JOIN vtxo ON payment.id=vtxo.payment_id
-WHERE round.txid = ?;
-`
-
-	selectSweepableRounds = `
-SELECT round.id, round.starting_timestamp, round.ending_timestamp, round.ended, round.failed, round.stage_code, round.txid, round.unsigned_tx, round.congestion_tree, round.forfeit_txs, round.connectors, round.connector_address, round.dust_amount, round.version, round.swept, payment.id, receiver.payment_id, receiver.pubkey, receiver.amount, receiver.onchain_address, vtxo.txid, vtxo.vout, vtxo.pubkey, vtxo.amount, vtxo.pool_tx, vtxo.spent_by, vtxo.spent, vtxo.redeemed, vtxo.swept, vtxo.expire_at, vtxo.payment_id
+	selectRound = `
+SELECT round.id, round.starting_timestamp, round.ending_timestamp, round.ended, round.failed, round.stage_code, round.txid, 
+round.unsigned_tx, round.connector_address, round.dust_amount, round.version, round.swept, payment.id, receiver.payment_id, 
+receiver.pubkey, receiver.amount, receiver.onchain_address, vtxo.txid, vtxo.vout, vtxo.pubkey, vtxo.amount, 
+vtxo.pool_tx, vtxo.spent_by, vtxo.spent, vtxo.redeemed, vtxo.swept, vtxo.expire_at, vtxo.payment_id, 
+tx.tx, tx.type, tx.position, tx.txid, 
+tx.tree_level, tx.parent_txid, tx.is_leaf
```

Comment:
do we really need ALL of theese fields in every repo fetch? I do think we can have multiple repo methods mapped to different subset of fields, to be mor efficient to only query the minimum data that you need for the method

Reply:
I agree, however it needs a complete refactor of the interface (all getters return the whole Round object), it also means rewrite badger implementation and tests. Let's do that in a dedicated PR no ?

Reply:
Open an issue about it! Will likely be a easy optimization